test/px-data-grid-paginated-test.js

```diff
@@ -131,6 +138,25 @@ document.addEventListener('WebComponentsReady', () => {
       }
     });
 
+    it('should hide page size selector if auto-hide enabled and not enough rows for multiple pages', (done) => {
+      grid.tableData = tableData; // 5 rows
+      grid.autoHidePageSizeSelect = true;
+      const pageSizes = [5, 10, 20];
+      navigation.selectablePageSizes = pageSizes;
+      Polymer.flush();
+      setTimeout(() => expect(pageSizeSelector).to.not.be.visible);
```

Comment:
This should be:

```javascript
setTimeout(() => {
  expect(pageSizeSelector).to.not.be.visible;
  done();
})
```

Otherwise `done()` will fire before the test assertions are evaluated.

Same treatment requires on lines 157-158.

Reply:
Thanks, fixed it - it seems setTimeout is not needed here at all. 
In addition to that, while debugging and fixing the tests, fixed a couple of other bugs both in the change and in the test code.